ethtxn/ethtxn.go

```diff
@@ -83,14 +92,52 @@ func NewTransaction(ctx context.Context, provider *ethrpc.Provider, txnRequest *
 		txnRequest.GasLimit = gasLimit
 	}
 
+	if txnRequest.To == nil && (txnRequest.Data == nil || len(txnRequest.Data) == 0) {
```

Comment:
I don't think you need `txnRequest.Data == nil` if you're already checking `len(txnRequest.Data) == 0` here.